### ISSUE TITLE
S1172 FP: Parameter not invoked but passed to other method

### ISSUE THREAD

**Issue by yanickSMA** [SonarSource/sonar-dotnet#5145]
### Description

A [S1172](https://rules.sonarsource.com/csharp/RSPEC-1172) warning is shown for a parameter used in a method if it's passed to another method without invoking it.

### Repro steps

In this code, there is a warning shown that the parameter nodes is unused
```csharp
static class TestProgram {
        static void Main(string[] args) {
            var dict = new Dictionary<string, int> {
                { "test", 1 }
            };
            Console.WriteLine(GetSegmentSortKey(dict));
            Console.ReadKey();
        }

        private static int? GetSegmentSortKey(IDictionary<string, int> nodes) {
            return Run(nodes.TryGetValueOrNull);
        }

        private static int? Run(Func<string, int?> tryGetValueOrNull) {
            return tryGetValueOrNull("test");
        }
    }

    public static class DictionaryExtensions {
        public static TValue? TryGetValueOrNull<TKey, TValue>(this IDictionary<TKey, TValue> dictionary, TKey key) where TValue : struct {
            if (dictionary != null && dictionary.TryGetValue(key, out var value)) {
                return value;
            } else {
                return null;
            }
        }
    }
```

This code functions the same but won't raise an errorCode

```csharp
static class TestProgram {
        static void Main(string[] args) {
            var dict = new Dictionary<string, int> {
                { "test", 1 }
            };
            Console.WriteLine(GetSegmentSortKey(dict));
            Console.ReadKey();
        }

        private static int? GetSegmentSortKey(IDictionary<string, int> nodes) {
            return Run(value => nodes.TryGetValueOrNull(value));
        }

        private static int? Run(Func<string, int?> tryGetValueOrNull) {
            return tryGetValueOrNull("test");
        }
    }

    public static class DictionaryExtensions {
        public static TValue? TryGetValueOrNull<TKey, TValue>(this IDictionary<TKey, TValue> dictionary, TKey key) where TValue : struct {
            if (dictionary != null && dictionary.TryGetValue(key, out var value)) {
                return value;
            } else {
                return null;
            }
        }
    }
```

### Expected behavior

Both code snippets should not raise the unused parameter warning.

### Actual behavior

In the first snippet the parameter nodes in GetSegmentSortKey is not detected and a warning for unused parameter is shown.

### Known workarounds

Insert method as a lambda

### Related information

* Visual Studio 16.11.5
* MSBuild 4.7.2
* SonarAnalyzer.CSharp 8.32.0.395
* Operating System Windows 10 Build 19042

**Comment by costin-zaharia-sonarsource** [comment]
Hi @yanickSMA, I can confirm the false positive. Thanks for reporting the issue.

**Comment by chrgrwilde** [comment]
@costin-zaharia-sonarsource Thanks for creating a repro and prioritizing this issue. Do you see any chance for this issue to get fixed in the near future?

**Comment by mary-georgiou-sonarsource** [comment]
Duplicate of https://github.com/SonarSource/sonar-dotnet/issues/5338
